conan/tools/cmake/cmakedeps/cmakedeps.py

```diff
@@ -112,3 +113,41 @@ def _generate_files(self, require, dep, ret, find_module_mode):
         # file is common for the different configurations.
         if not os.path.exists(config.filename):
             ret[config.filename] = config.render()
+
+    def set_property(self, dep, prop, value, build_context=False):
+        build_suffix = "_build" if build_context else ""
```

Comment:
```suggestion
        build_suffix = "&build" if build_context else ""
```

Better use something that is an invalid package name, because if some user has a "host" package called ``mypkg_build`` and a tool called ``mypkg``, this will error.